File: 383 ransom note/solution.cpp
#include <vector>
#include <string>

class Solution {
public:
    bool canConstruct(std::string ransomNote, std::string magazine) {
        std::vector<int> letters(255, 0);
        for (char l : magazine)
            letters[l]++;
        for (char l : ransomNote){
            if (letters[l] == 0){
                return false;
            }
            letters[l]--;
        }
        return true;
    }
};

File: 383 ransom note/solution.py
from collections import Counter

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        letters = Counter(magazine)
        note = Counter(ransomNote)

        for key in note.keys():
            if key not in letters.keys():
                return False
            if note[key] > letters[key]:
                return False
        return True